.ci-builds/.azure-pipelines.yml

```diff
@@ -159,31 +136,79 @@ jobs:
         CXX: clang++
         LANG: "en_US@calendar=gregorian;hours=h12"
 #-------------------------------------------------------------------------
-- job: ICU4C_Clang13_Cpp20_Ubuntu_2004
-  displayName: 'C: Linux Clang-13 C++20 (Ubuntu 20.04)'
+- job: ICU4C_Clang14_Ubuntu_2004_WarningsAsErrors
+  displayName: 'C: Linux Clang-14 WarningsAsErrors (Ubuntu 20.04)'
+  timeoutInMinutes: 30
+  pool:
+    vmImage: 'ubuntu-20.04'
+  steps:
+    - checkout: self
+      lfs: true
+      fetchDepth: 10
+    # Install Clang-14 from https://apt.llvm.org/
+    - script: |
+        curl -Ls https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
+        sudo apt-add-repository 'deb http://apt.llvm.org/focal/ llvm-toolchain-focal-14 main'
+        sudo apt update
+        sudo apt install -qy clang-14 lld-14 libc++-14-dev libc++abi-14-dev
+      displayName: 'Install Clang-14'
+    - script: |
+        export CPPFLAGS="-Werror -Wall -Wextra -Wextra-semi -Wundef -Wnon-virtual-dtor -Wsuggest-override" && cd icu4c/source && ./runConfigureICU Linux && make -j2 tests
```

Comment:
optional: pull the CPPFLAGS out into "env", like for some of the other bots?

Reply:
Sure, will update.